Result/Result.swift

```diff
@@ -31,7 +31,10 @@ public enum Result<T, Error: Swift.Error>: ResultProtocol, CustomStringConvertib
 	public init(attempt f: () throws -> T) {
 		do {
 			self = .success(try f())
-		} catch {
+		} catch var error {
+			if Error.self == AnyError.self {
```

Comment:
I think that is already covered here: https://github.com/antitypical/Result/blob/65d0c81e295e3781c9fea5bec309be049b045a3a/Result/Result.swift#L183-L187